homeassistant/components/sensor/openhardwaremonitor.py

```diff
@@ -0,0 +1,192 @@
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.value
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sun."""
+        return self.attributes
+
+    def update(self):
+        """Update the device from a new JSON object."""
+        self._data.update()
+
+        array = self._data.data[OHM_CHILDREN]
+        _attributes = {}
+
+        for path_index in range(0, len(self.path)):
+            path_number = self.path[path_index]
+            values = array[path_number]
+
+            if path_index == len(self.path) - 1:
+                self.value = values[OHM_VALUE].split(' ')[0]
+                _attributes.update({
+                    'name': values[OHM_NAME],
+                    STATE_MIN_VALUE: values[OHM_MIN].split(' ')[0],
+                    STATE_MAX_VALUE: values[OHM_MAX].split(' ')[0]
+                })
+
+                self.attributes = _attributes
+                return
+            else:
+                array = array[path_number][OHM_CHILDREN]
+                _attributes.update({
+                    'level_%s' % path_index: values[OHM_NAME]
+                })
+
+
+class OpenHardwareMonitorData(object):
+    """Class used to pull data from OHM and create sensors."""
+
+    def __init__(self, config, hass):
+        """Initialize the Open Hardware Monitor data-handler."""
+        self.data = None
+        self._config = config
+        self._hass = hass
+        self.devices = []
+        self.initialize(utcnow())
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Hit by the timer with the configured interval."""
+        if self.data is None:
+            self.initialize(utcnow())
+        else:
+            self.refresh()
+
+    def refresh(self):
+        """Download and parse JSON from OHM."""
+        data_url = "http://%s:%d/data.json" % (
+            self._config.get(CONF_HOST),
+            self._config.get(CONF_PORT))
+
+        try:
+            response = requests.get(data_url)
+            self.data = response.json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is OpenHardwareMonitor running?")
+
+    def schedule_retry(self):
+        """Schedule a retry in 30 seconds."""
+        _LOGGER.info("Retrying in 30 seconds")
+
+        async_track_point_in_utc_time(
+            self._hass, self.initialize, utcnow() + RETRY_INTERVAL)
+
+    def initialize(self, now):
+        """Initial parsing of the sensors and adding of devices."""
+        self.refresh()
+
+        if self.data is None:
+            self.schedule_retry()
+            return
+
+        self.devices = self.parse_children(self.data, [], [], [])
+
+    def parse_children(self, json, devices, path, names):
+        """Recursively loop through child objects, finding the values."""
+        result = devices.copy()
+
+        if len(json[OHM_CHILDREN]) > 0:
```

Comment:
It actually checks 2 different properties of the `json` object. First it checks if `OHM_CHILDREN` is an array, *if not* then it will check if the `OHM_VALUE` contains at least a space. The format of Open Hardware Monitor is pretty stable in this, but we don't want to be adding sensors that do not have the **`[VALUE] [UNIT_OF_VALUE]`** format.